data/modules/TradeShips/Debug.lua

```diff
@@ -0,0 +1,333 @@
+						},{
+							callbacks = {
+								onClick = function(row)
+									Game.sectorView:SwitchToPath(row.from)
+								end,
+								isSelected = function(row)
+									return row.from:IsSameSystem(selected_in_sectorview)
+								end
+						}})
+					ui.treePop()
+				end
+			end
+		end
+
+		if ui.collapsingHeader("All ships") then
+			local ships = {}
+			for ship, trader in pairs(Core.ships) do
+				table.insert(ships,
+					{
+						ship = ship,
+						label = ship:GetLabel(),
+						status = trader.status,
+						ts_error = trader.ts_error,
+						cargo = ship.usedCargo,
+						ai = ship:GetCurrentAICommand(),
+						model = ship.shipId,
+						port = trader.starport and trader.starport:GetLabel()
+					})
+			end
+			local obj = Game.systemView:GetSelectedObject()
+			local ship_selected = obj.type == Engine.GetEnumValue("ProjectableTypes", "OBJECT") and obj.base == Engine.GetEnumValue("ProjectableBases", "SHIP")
+			drawArrayTable("tradeships_all", ships, ipairs,
+				{
+					{ name = "#",      key = "#"                       },
+					{ name = "Label",  key = "label",    string = true },
+					{ name = "Model",  key = "model",    string = true },
+					{ name = "Status", key = "status",   string = true },
+					{ name = "Error",  key = "ts_error", string = true },
+					{ name = "Cargo",  key = "cargo"                   },
+					{ name = "AI",     key = "ai",       string = true },
+					{ name = "Port",   key = "port",     string = true }
+				},
+				{ callbacks = {
+						onClick =	function(row)
+							if row.status ~= "hyperspace" and row.status ~= "hyperspace_out" then
+								Game.systemView:SetSelectedObject(Engine.GetEnumValue("ProjectableTypes", "OBJECT"),
+									Engine.GetEnumValue("ProjectableBases", "SHIP"), row.ship)
+							end
+						end,
+						isSelected = function(row)
+							return ship_selected and Game.systemView:GetSelectedObject().ref == row.ship
+						end
+					}
+				})
+		end
+		if ui.collapsingHeader("Log") then
+			local obj = Game.systemView:GetSelectedObject()
+			local ship_selected = obj.type == Engine.GetEnumValue("ProjectableTypes", "OBJECT") and obj.base == Engine.GetEnumValue("ProjectableBases", "SHIP")
+			search_text, _ = ui.inputText("Search log", search_text, {})
+			drawArrayTable("tradeships_log", Core.log, Core.log.iter(search_text ~= "" and function (row)
+				return
+				row.label and string.match(row.label, search_text) or
+				row.msg and string.match(row.msg, search_text) or
+				row.model and string.match(row.model, search_text)
+			end),{
+				{ name = "Time",    key = "time" , fnc = ui.Format.Datetime                },
+				{ name = "Label",   key = "label",                           string = true },
+				{ name = "Model",   key = "model",                           string = true },
+				{ name = "Message", key = "msg",                             string = true }
+			},{
+				callbacks = {
+					onClick =	function(row)
```

Comment:
Extra tab between `=` and `function`. There are a few more of these that I'm not going to individually point out.

Reply:
Done!